dotCMS/src/main/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResource.java

```diff
@@ -0,0 +1,217 @@
+            final List<ServiceIntegrationView> integrationViews = helper.getAvailableDescriptorViews(user);
+            if(!integrationViews.isEmpty()){
+                return Response.ok(new ResponseEntityView(integrationViews)).build(); // 200
+            }
+            throw new DoesNotExistException("Nope. No service integration is available to configure. Try uploading a file!");
+        } catch (Exception e) {
+            Logger.error(this.getClass(), "Exception on listing available service integration descriptors.", e);
+            return ResponseUtil.mapExceptionResponse(e);
+        }
+    }
+
+    @GET
+    @Path("/{serviceKey}")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public final Response getServiceIntegrationByKey(
+            @Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @PathParam("serviceKey") final String serviceKey,
+            @QueryParam("hostId") final String hostId
+    ) {
+        try {
+            final InitDataObject initData =
+                    new WebResource.InitBuilder(webResource)
+                            .requiredBackendUser(true)
+                            .requiredFrontendUser(false)
+                            .requestAndResponse(request, response)
+                            .rejectWhenNoUser(true)
+                            .init();
+            final User user = initData.getUser();
+            if (UtilMethods.isSet(hostId)) {
+                //if the hostId is set it will be used to bring the configuration detail for that specific host
+                final Optional<ServiceIntegrationDetailedView> serviceIntegrationDetailedView = helper
+                        .getServiceIntegrationHostDetailedView(serviceKey, hostId, user);
+                if (serviceIntegrationDetailedView.isPresent()) {
+                    return Response.ok(new ResponseEntityView(serviceIntegrationDetailedView.get()))
+                            .build(); // 200
+                }
+                throw new DoesNotExistException(String.format(
+                        "Nope. No service integration was found for key `%s` and hostId `%s`. ",
+                        serviceKey, hostId));
+            }
+            final Optional<ServiceIntegrationHostView> serviceIntegrationView = helper
+                    .getServiceIntegrationHostView(serviceKey, user);
+            if (serviceIntegrationView.isPresent()) {
+                return Response.ok(new ResponseEntityView(serviceIntegrationView.get()))
+                        .build(); // 200
+            }
+            throw new DoesNotExistException(
+                    String.format("Nope. No service integration was found for key `%s`. ",
+                            serviceKey));
+        } catch (Exception e) {
+            Logger.error(this.getClass(), "Exception on blah blah blah message: " + e.getMessage(),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "Exception on blah blah blah message: " appears 4 times in this file; the first occurrence is on line 122](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)